### PR TITLE
Fixing classpath for Java11 support in map reduce launcher

### PR DESCRIPTION
**Pull request by iam-divya**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java [hunk 1]
```diff
@@ -123,4 +127,11 @@ public void run() {
       throw t;
     }
   }
+  private static URL getURLByClass(ClassLoader classLoader, String className) {
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java:130]: checkstyle

**Review by albertshau — APPROVED** [review]
lgtm after checkstyle fixes

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java [hunk 2]
```diff
@@ -58,8 +60,10 @@ public static void launch(String mainClassName, String[] args) throws Exception
       throw new IllegalStateException("Failed to find resource for main class " + mainClassName);
     }
 
-    if (!urls.remove(ClassLoaders.getClassPathURL(mainClassName, resource))) {
-      throw new IllegalStateException("Failed to remove main class resource " + resource);
+    URL urlByClass = getURLByClass(systemClassLoader, mainClassName);
+    if (!urls.remove(urlByClass)) {
+      urls.forEach(url -> LOG.info("URL: {}", url));
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java]: Generally, don’t log and throw.

**Review by chtyim — APPROVED** [review]

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java (hunk 2 above)
Comment (iam-divya) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java]: Updated

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java (hunk 1 above)
Comment (iam-divya) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java:130]: Fixed

**Pull request merged**